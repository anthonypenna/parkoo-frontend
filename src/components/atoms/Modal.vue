<template>
  <Fade>
    <div class="modal">
      <slot />
    </div>
  </Fade>
</template>

<script lang="ts">
import { Map } from "mapbox-gl";
import Vue from "vue";
import Fade from "@/components/atoms/Fade.vue";
import { mapGetters } from "vuex";

export default Vue.extend({
  name: "Modal",

  components: { Fade },

  computed: {
    ...mapGetters("mapbox", ["map"]),
  },

  mounted() {
    const map = this.map as Map;
    map.dragPan.disable();
    map.scrollZoom.disable();
  },

  destroyed() {
    const map = this.map as Map;
    map.dragPan.enable();
    map.scrollZoom.enable();
  },
});
</script>

<style lang="scss" scoped>
.modal {
  position: fixed;
  top: 50%;
  left: 50%;
  max-width: 560px;
  width: calc(100% - (56px + 2rem));
  padding: 32px 28px 16px;
  background: #fff;
  border-radius: 15px;
  box-shadow: 0 4px 16px rgba($color: #000000, $alpha: 0.1);
  transform: translate(-50%, -50%);
  z-index: 10;
}
</style>
