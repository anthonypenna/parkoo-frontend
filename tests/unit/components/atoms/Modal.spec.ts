import { createLocalVue, shallowMount } from '@vue/test-utils'
import Modal from '@/components/atoms/Modal.vue'
import Vuex from 'vuex'

const localVue = createLocalVue()

localVue.use(Vuex)

const store = new Vuex.Store({
  modules: {
    mapbox: {
      namespaced: true,
      state: {
        map: {
          dragPan: {
            disable: jest.fn(),
            enable: jest.fn(),
          },
          scrollZoom: {
            disable: jest.fn(),
            enable: jest.fn(),
          },
        },
      },
      getters: {
        map: state => state.map,
      },
    },
  },
})

describe('<Modal />', () => {
  it('should render correctly', () => {
    const wrapper = shallowMount(Modal, { localVue, store })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render the correct slot', () => {
    const wrapper = shallowMount(Modal, {
      localVue,
      store,
      slots: { default: 'Hello world!' },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
