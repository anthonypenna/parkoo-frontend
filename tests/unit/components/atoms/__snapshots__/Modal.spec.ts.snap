// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render correctly 1`] = `
<fade-stub>
  <div class="modal"></div>
</fade-stub>
`;

exports[`<Modal /> should render the correct slot 1`] = `
<fade-stub>
  <div class="modal">Hello world!</div>
</fade-stub>
`;
